semgrep-core/src/parsing/pfff/scala_to_generic.ml

```diff
@@ -562,24 +562,7 @@ and v_stmt = function
       let v1 = v_tok v1
       and v2 = v2 |> G.unbracket |> v_enumerators
       and v3 = v_for_body v3 in
-      List.fold_right
-        (fun gen stmt ->
-          match gen with
-          | `G (pat, tok, e, guards) ->
-              G.For
-                ( v1,
-                  G.ForEach (pat, tok, e),
-                  List.fold_right
-                    (fun (g_tok, g_e) stmt ->
-                      G.If (g_tok, G.Cond g_e, stmt, None) |> G.s)
-                    guards stmt )
-              |> G.s
-          | `GIf guards ->
-              List.fold_right
-                (fun (g_tok, g_e) stmt ->
-                  G.If (g_tok, G.Cond g_e, stmt, None) |> G.s)
-                guards stmt)
-        v2 v3
+      G.For (v1, G.MultiForEach v2, v3) |> G.s
```

Comment:
I usually try to avoid adding a construct in AST_generic when it's useful only for one language, but
maybe it's ok for this one indeed.